e2e/playground.py

```diff
@@ -0,0 +1,2135 @@
+    ) -> None:
+        if isinstance(value, list):
+            assert len(value) > 0, "`value` must be `None` or a non-empty list"
+        value_str = "null" if value is None else json.dumps(value)
+        expect_attr(
+            self.loc,
+            "data-date-dates-disabled",
+            value=value_str,
+            timeout=timeout,
+        )
+
+    def expect_daysofweekdisabled(
+        self,
+        value: typing.Union[typing.List[int], None],
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        if isinstance(value, list):
+            assert len(value) > 0, "`value` must be `None` or a non-empty list"
+        value_str = "null" if value is None else json.dumps(value)
+        expect_attr(
+            self.loc,
+            "data-date-days-of-week-disabled",
+            value=value_str,
+            timeout=timeout,
+        )
+
+
+class InputDate(_DateBase):
+    def __init__(self, page: Page, id: str) -> None:
+        super().__init__(
+            page,
+            id=id,
+            loc="input[type=text].form-control",
+            loc_label=f"label#{id}-label",
+            loc_container=f"div#{id}.shiny-input-container",
+        )
+
+
+class InputDateRange(_WidthContainer, _InputWithLabel):
+    # id: str,
+    # label: TagChildArg,
+    # *,
+    # start: Optional[Union[date, str]] = None,
+    # end: Optional[Union[date, str]] = None,
+    # min: Optional[Union[date, str]] = None,
+    # max: Optional[Union[date, str]] = None,
+    # format: str = "yyyy-mm-dd",
+    # startview: str = "month",
+    # weekstart: int = 0,
+    # language: str = "en",
+    # separator: str = " to ",
+    # width: Optional[str] = None,
+    # autoclose: bool = True,
+
+    loc_separator: Locator
+    loc_start: Locator
+    loc_end: Locator
+    date_start: _DateBase
+    date_end: _DateBase
+
+    def __init__(self, page: Page, id: str) -> None:
+        super().__init__(
+            page,
+            id=id,
+            loc="input[type=text].form-control",
+            loc_label=f"label#{id}-label",
+            loc_container=f"div#{id}.shiny-input-container",
+        )
+        self.loc_separator = self.loc_container.locator(".input-daterange > span")
+        self.loc_start = self.loc.nth(0)
+        self.loc_end = self.loc.nth(1)
+        self.date_start = _DateBase(
+            page,
+            id=id,
+            loc=self.loc_start,
+            loc_label=self.loc_label,
+            loc_container=self.loc_container,
+        )
+        self.date_end = _DateBase(
+            page,
+            id=id,
+            loc=self.loc_end,
+            loc_label=self.loc_label,
+            loc_container=self.loc_container,
+        )
+
+    def set(
+        self,
+        # TODO-barret; Should this be a list or a tuple?
+        value: typing.Union[
+            typing.Tuple[
+                typing.Union[datetime.date, str, None],
+                typing.Union[datetime.date, str, None],
```

Comment:
1. Don't support `datetime.date` (because of the language parameter).
2. Make `None` a no-op.

Reply:
806a064